pkg/liveshare/port_forwarder.go

```diff
@@ -5,9 +5,15 @@ import (
 	"fmt"
 	"io"
 	"net"
+	"time"
 
 	"github.com/opentracing/opentracing-go"
 	"golang.org/x/crypto/ssh"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	connectionTimeout = 30 * time.Second
```

Comment:
Port forwarder is a basic primitive, let callers set their timeouts, we just enforce it.